internal/provider/privateinternetaccess/tokenutils.go

```diff
@@ -0,0 +1,72 @@
+package privateinternetaccess
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func fetchToken(ctx context.Context, client *http.Client,
+	tokenType string, authFilePath string) (token string, err error) {
```

Comment:
since this is used outside openvpn-only, we should inject the username and password and not use the openvpn auth file anymore (especially if it's for Wireguard, that makes it super strange)